lab_5_scrapper/scrapper.py

```diff
@@ -120,46 +261,207 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
+        regular = re.match(r"\d{1,2} [A-Za-z]+ \d{4}, \d{1,2}:\d{2}", date_str)
+        if regular:
+            return datetime.datetime.strptime(date_str, '%d %B %Y, %H:%M')
+        return datetime.datetime.strptime(date_str, '%H:%M, %d %B %Y')
+
+
 
     def parse(self) -> Union[Article, bool, list]:
         """
         Parses each article
         """
-        pass
+        response = make_request(self.article_url, self.config)
+        article_soup = BeautifulSoup(response.text, 'html.parser')
+        self._fill_article_with_text(article_soup)
+        self._fill_article_with_meta_information(article_soup)
+        return self.article
+
 
 
 def prepare_environment(base_path: Union[Path, str]) -> None:
     """
     Creates ASSETS_PATH folder if no created and removes existing folder
     """
-    pass
+    if base_path.exists():
+        shutil.rmtree(base_path)
+    base_path.mkdir(parents=True)
+
+
+class CrawlerRecursive(Crawler):
+    """
+    Recursive Crawler implementation
+    """
+
+    def __init__(self, config: Config) -> None:
+        """
+        Initializes an instance of the Recursive Crawler class
+        """
+        super().__init__(config)
+        self.start_url = self.get_search_urls()[0]
+        self.urls = []
+        self.crawler_data_path = Path('crawler_sta.json')
+        if self.crawler_data_path.exists():
+            self.load_crawler_data()
+
+    def save_crawler_data(self) -> None:
+        """
+        Saves values of start_url and urls attributes
+        in a json file
+        """
+        info_dict = {'start_url': self.start_url, 'urls': self.urls}
+        with self.crawler_data_path.open('w', encoding='utf-8') as json_file:
+            json.dump(info_dict, json_file, ensure_ascii=False, indent=4)
+
+    def load_crawler_data(self) -> None:
+        """
+        Loads start_url and urls values, saved before interruption,
+        from json file
+        """
+        if self.crawler_data_path.exists() and self.crawler_data_path.stat().st_size != 0:
+            with self.crawler_data_path.open('r', encoding='utf-8') as json_file:
+                info_dict = json.load(json_file)
+            self.start_url = info_dict['start_url']
+            self.urls = info_dict['urls']
+
+    def find_articles(self) -> None:
+        """
+        Finds articles recursively starting from the given URL
+        """
+        if len(self.urls) >= self.config.get_num_articles():
+            return
+
+        response = make_request(self.start_url, self.config)
+        article_bs = BeautifulSoup(response.text, 'html.parser')
+        urls = article_bs.select('div.mininews') + article_bs.select('div.midinews')
+
+        for url in urls:
+            article_url = self._extract_url(url)
+            if not article_url or article_url in self.urls:
+                continue
+
+            self.urls.append(article_url)
+            self.start_url = article_url
```

Comment:
something is principally wrong. you need to 

1. start with domain name of website
2. visit any page of website you get during crawling, keep them to avoid cycling 
3. keep article urls as this is our goal
4. stop when articles are found in requested volume